### ISSUE TITLE
HTTP Requests based API. Google Vision API. Proxy.

### ISSUE THREAD

**Issue by zhukovleonid121** [MatheusVp2/hCaptcha-Solver-MR#2]
Hello. Thanks for your work, everything works great. There are only a few chips missing for a complete picture. Is it possible to add an http-based server to the soft api so that any software can send a request for captcha solutions and receive a response in the form of a key after some time? Suppose I send a request to the server to solve the captcha with the desired host and sitekey, receive the ID from the server and after the next request every 5-10 seconds I check whether the captcha solution is ready, if it is ready, I receive the key in the response. As I understand it, GO has already implemented this, it would be cool to see it in python. (https://github.com/JustTalDevelops/hcaptcha-solver-api)
It would also be great to add a captcha solution via the Google Vision API as an alternative to yolo, it would improve performance on weak PCs. Thank you !

**Comment by MatheusVp2** [comment]
Thanks for the feedback, I understood the information about creating an API, I can implement and inform when to update, at the moment I'm full of work in college, I will try soon. Is Google vision free? Could you help me with it to implement?
I was thinking about downloading the images that are used in the captchas and creating an internal training of the images to increase the faster hits of the captchas!

**Comment by zhukovleonid121** [comment]
> Thanks for the feedback, I understood the information about creating an API, I can implement and inform when to update, at the moment I'm full of work in college, I will try soon. Is Google vision free? Could you help me with it to implement?
> I was thinking about downloading the images that are used in the captchas and creating an internal training of the images to increase the faster hits of the captchas!

Thank you for your reply. Google vision is free if you get a 300$ credit at cloud.google.com. I've tried it several times already, it requires paperwork sometimes and a real credit card is needed. I used to get it easily, but now there are difficulties. I'll try to get it again, if I can, I'll let you know.
Into the proxy in the software. I kind of saw its implementation in the very first version of the hcaptcha solver (https://github.com/BoicCodes/hCaptcha-solver/blob/master/main.py). But it seems like you can't insert a proxy with authorization here and they don't mix.
Btw, there is one more note on hcaptcha. Let's say hcaptcha is embedded in the discord.com mobile application at login, but if I even call it 50 times in a row from one IP address, it still gives me a stably valid token WITHOUT solving the pictures. It's amazing and weird. And Hcaptcha almost never displays pictures through a regular chrome browser when logging in. I tried to substitute headers from the application, but it was unsuccessful, in the software it always gives pictures. Perhaps you need some kind of cookies, but they seem to be missing in the mobile application..

**Comment by MatheusVp2** [comment]
Yes, but in this one you saw, it kind of does web scraping on the page, mine gets straight from the server of the Hcaptcha API routes, doing the same process that it does behind the web!
Then I do it step by step, and put an image detector to answer the photos!
You can put a proxy on the requests you do too!
